stores/metadata.go

```diff
@@ -397,6 +410,89 @@ func (raw rawObject) toSlabSlice() (slice object.SlabSlice, _ error) {
 	return slice, nil
 }
 
+func (s *SQLStore) Bucket(_ context.Context, bucket string) (api.Bucket, error) {
+	var b dbBucket
+	err := s.db.
+		Model(&dbBucket{}).
+		Where("name = ?", bucket).
+		Take(&b).
+		Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return api.Bucket{}, api.ErrBucketNotFound
+	} else if err != nil {
+		return api.Bucket{}, err
+	}
+	return api.Bucket{
+		CreationDate: b.CreatedAt.UTC(),
+		Name:         b.Name,
+	}, nil
+}
+
+func (s *SQLStore) CreateBucket(_ context.Context, bucket string) error {
+	// Create bucket.
+	return s.retryTransaction(func(tx *gorm.DB) error {
+		res := tx.Clauses(clause.OnConflict{
+			DoNothing: true,
+		}).
+			Create(&dbBucket{Name: bucket})
+		if res.Error != nil {
+			return res.Error
+		} else if res.RowsAffected == 0 {
+			return api.ErrBucketExists
+		}
+		return nil
+	})
+}
+
+func (s *SQLStore) DeleteBucket(_ context.Context, bucket string) error {
+	// Delete bucket.
+	return s.retryTransaction(func(tx *gorm.DB) error {
+		var b dbBucket
+		if err := tx.Take(&b, "name = ?", bucket).Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			return api.ErrBucketNotFound
+		} else if err != nil {
+			return err
+		}
+		var count int64
+		if err := tx.Model(&dbObject{}).Where("db_bucket_id = ?", b.ID).
+			Limit(1).
+			Count(&count).Error; err != nil {
+			return err
+		}
+		if count > 0 {
+			return api.ErrBucketNotEmpty
+		}
+		res := tx.Where("buckets.id = ?", b.ID).
```

Comment:
we can reuse `var b dbBucket` here I think and prevent doing the `ErrBucketNotFound` twice